src/shims/env.rs

```diff
@@ -310,6 +338,27 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
+    #[allow(non_snake_case)]
+    fn SetCurrentDirectoryW (
+        &mut self,
+        path_op: OpTy<'tcx, Tag> // LPCTSTR
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+        this.assert_target_os("windows", "SetCurrentDirectoryW");
+
+        this.check_no_isolation("SetCurrentDirectoryW")?;
+
+        let path = this.read_path_from_wide_str(this.read_scalar(path_op)?.not_undef()?)?;
+
+        match env::set_current_dir(path) {
+            Ok(()) => Ok(1),
+            Err(e) => {
+                this.set_last_error_from_io_error(e)?;
```

Comment:
Is `set_last_error_from_io_error` prepared for being called on Windows?

Reply:
It is not yet fully prepared to be called for Windows targets.. I made a rebase push that includes partial implementation of `set_last_error_from_io_error` for Windows targets. For now, the PR passes test `tests/run-pass/current_dir.rs`.